src/target/AbstractTarget.php

```diff
@@ -179,4 +198,9 @@ public static function any(): self
 
         return static::$anyTarget;
     }
+
+    public function isDeleted(): bool
+    {
+        return $this->getState() === 'deleted';
```

Comment:
This library works on abstraction only. Concrete state names (the same as types) should not appear in this repo